bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -178,9 +261,39 @@ public void propertyChange(PropertyChangeEvent evt) {
 			}
 
 			// Sky brightness warnings
+
 			Double sb = a.getMin(Circumstance.TOTAL_SKY_BRIGHTNESS, false);
-			if (sb != null && !a.getObs().getConditions().containsSkyBrightness(sb))
+
+			final Option<Interval> skyBackgroundInterval = night.flatMap(n -> {
+				final Obs obs = a.getObs();
+				final SkyBackgroundSolver skyBackgroundSolver = new SkyBackgroundSolver(site, obs::getCoords, sb);
+				final SortedSet<Interval> intervals = skyBackgroundSolver.solve(n).getIntervals();
+				if (intervals.isEmpty()) {
+					return None.instance();
+				} else {
+					final Interval interval = intervals.first();
+					// For some reason the default background interval is larger than the night interval
+					if (interval.getStart() >= n.getStart() && interval.getEnd() <= n.getEnd()) {
+						return None.instance();
+					}
+					return new Some<>(intervals.first());
+				}});
+
+			if (sb != null && !a.getObs().getConditions().containsSkyBrightness(sb)) {
 				mm.addMarker(false, this, Severity.Error, "Sky brightness constraint violated.", v, a);
+			} else if (!skyBackgroundInterval.isEmpty()) {
+				mm.addMarker(
+					false,
+					this,
+					Severity.Notice,
+					String.format(
+						"Must be observed in the sky background range %s - %s",
+						dateFormat(zone).format(Instant.ofEpochMilli(skyBackgroundInterval.getValue().getStart())),
+						dateFormat(zone).format(Instant.ofEpochMilli(skyBackgroundInterval.getValue().getEnd()))
```

Comment:
Wouldn't you want to use your `formatInterval` method here as well?  Is it just that parens would be used?  That seems like it would be okay too.

Reply:
It seems like what we would really like, instead of my above solution, is something along the line of:

```
final String formatInterval(final TimeZone zone, final Interval interval, final String sep = " - ") {
    final Function<Long, String> f = ms ->
        DateTimeFormatter.ofPattern("HH:mm").withZone(zone.toZoneId()).format(Instant.ofEpochMilli(ms));
   return f.apply(interval.getBegin()) + set + f.apply(interval.getEnd());
```
All  other bells and whistles (such as surrounding parentheses) can be done outside in the calling code of this method.


Then  L289 - 29 changes to:
```
"Must be observed in the sky background rage " + formatInterval(zone, skyBackgroundInterval.getValue());
```
Although using the naive exception-throwing `SkyBackground.getValue` makes me uncomfortable still.